openmc/universe.py

```diff
@@ -710,6 +716,50 @@ def get_all_cells(self, memo=None):
     def get_all_materials(self, memo=None):
         return OrderedDict()
 
+    def _n_geom_elements(self, geom_type):
+        """
+        Helper function for retrieving the number geometric entities in a DAGMC
+        file
+
+        Parameters
+        ----------
+        geom_type : str
+            The type of geometric entity to count. One of {'Volume', 'Surface'}. Returns
+            the runtime number of voumes in the DAGMC model (includes implicit complement).
+
+        Returns
+        -------
+        int
+            Number of geometry elements of the specified type
+        """
+        cv.check_value('geometry type', geom_type, ('volume', 'surface'))
+
+        def decode_str_tag(tag_val):
+            return tag_val.tobytes().decode().replace('\x00', '')
+
+        with h5py.File(self.filename) as dagmc_file:
+            category_data = dagmc_file['tstt/tags/CATEGORY/values']
+            category_strs = map(decode_str_tag, category_data)
+            # add one assuming implicit complement doesn't exist
```

Comment:
```suggestion
```
If I've understood line 751 then I think the code checks for the impl_complement and doesn't assume it exists

Reply:
Whoops, yeah I changed my approach and forgot to remove this comment. Thanks!